tgui/packages/tgui/interfaces/RapidPipeDispenser.tsx

```diff
@@ -231,60 +219,66 @@ const LayerSelect = (props) => {
       </Button.Checkbox>
     </LabeledList.Item>
   );
+}
+
+type RecipeRowProps = {
+  recipe: Recipe;
+  shownCategory: Category;
 };
 
-const PreviewSelect = (props) => {
+function RecipeRow(props: RecipeRowProps) {
   const { act } = useBackend<Data>();
+  const { recipe, shownCategory } = props;
+
   return (
-    <Box>
-      {props.previews.map((preview) => (
-        <Button
-          ml={0}
-          key={preview.dir}
-          tooltip={preview.dir_name}
-          selected={preview.selected}
-          style={{
-            width: '40px',
-            height: '40px',
-            padding: '0',
-          }}
-          onClick={() => {
-            act('pipe_type', {
-              pipe_type: props.pipe_type,
-              category: props.category,
-            });
-            act('setdir', {
-              dir: preview.dir,
-              flipped: preview.flipped,
-            });
-          }}
-        >
-          <Box
-            className={classes([
-              'pipes32x32',
-              preview.dir + '-' + preview.icon_state,
-            ])}
-            style={{
-              transform: 'scale(1.5) translate(9.5%, 9.5%)',
+    <Stack fill>
+      {recipe.previews.map((preview) => (
+        <Stack.Item key={preview.dir}>
+          <ImageButton
+            asset={['pipes32x32', preview.dir + '-' + preview.icon_state]}
```

Comment:
```suggestion
            asset={['pipes32x32', `${preview.dir}-${preview.icon_state}`]}
```
I don't know how you feel about it, but I think it's better.

Reply:
added